lib/vagrant-libvirt/action/handle_box_image.rb

```diff
@@ -15,12 +15,62 @@ def initialize(app, _env)
           @app = app
         end
 
-        def call(env)
-          # Verify box metadata for mandatory values.
-          #
+        def self.get_volume_name(env, index)
+          name = env[:machine].box.name.to_s.dup.gsub('/', '-VAGRANTSLASH-')
+          name << "_vagrant_box_image_#{
+          begin
+            env[:machine].box.version.to_s
+          rescue
+            ''
+          end}_#{index}.img"
+          return name
+        end
+
+        def self.get_virtual_size(env)
           # Virtual size has to be set for allocating space in storage pool.
           box_virtual_size = env[:machine].box.metadata['virtual_size']
           raise Errors::NoBoxVirtualSizeSet if box_virtual_size.nil?
+          return box_virtual_size
+        end
+
+        def self.get_default_box_image_path(index)
+          return index <= 0 ? 'box.img' : "box_#{index}.img"
+        end
+
+        def self.get_box_image_path(env, box_name)
+          return env[:machine].box.directory.join(box_name).to_s
+        end
+
+        def self.verify_virtual_size_in_disks(disks)
+          disks.each_with_index do |disk, index|
+            raise Errors::NoDiskVirtualSizeSet, disk_index:index if disk['virtual_size'].nil?
+          end
+        end
+
+        def call(env)
+          # Verify box metadata for mandatory values.
+          #
+          # Verify disk number
+          disks = env[:machine].box.metadata.fetch('disks', [])
+          if disks.empty?
+            disks.push({
+              'path' => HandleBoxImage.get_default_box_image_path(0),
+              'name' => HandleBoxImage.get_volume_name(env, 0),
+              'virtual_size' => HandleBoxImage.get_virtual_size(env),
```

Comment:
I can't comment on it properly, but looking further down I see a `format` entry in the metadata that I think needs to be added here as it should be associated with individual disks.

Thoughts on whether there should be a top level value that is a default or should assume a default value of `qcow2`? Either way, might be better to decided whether it is handled now to be settable for each individual disk and whether it is therefore required to be specified for each disk or has a default.

Reply:
With the new function `send_box_image`, the `box_format` is a default value for each disks. If you want I may append the box format options for each disks.

Reply:
No need, if we're using that as a default for all disk images it should be fine.